core/src/main/scala/org/apache/spark/api/python/PythonUtils.scala

```diff
@@ -29,14 +29,9 @@ import org.apache.spark.api.java.{JavaRDD, JavaSparkContext}
 private[spark] object PythonUtils {
   val PY4J_ZIP_NAME = "py4j-0.10.9-src.zip"
 
-  /** Get the PYTHONPATH for PySpark, either from SPARK_HOME, if it is set, or from our JAR */
+  /** Get the PYTHONPATH for PySpark from our JAR */
   def sparkPythonPath: String = {
     val pythonPath = new ArrayBuffer[String]
-    for (sparkHome <- sys.env.get("SPARK_HOME")) {
-      pythonPath += Seq(sparkHome, "python", "lib", "pyspark.zip").mkString(File.separator)
-      pythonPath +=
-        Seq(sparkHome, "python", "lib", PY4J_ZIP_NAME).mkString(File.separator)
-    }
```

Comment:
Hey, I don't think we can remove this. I think you refer to Yarn cluster, right? What about standalone mode or other modes?

Reply:
Yes, this is for yarn mode. Could we determine whether this is yarn mode, then skip this pythonPath?

Reply:
I would just leave this code for simplicity.

> If local machine has different version of both files

How would this happen? pyspark.zip and py4j.zip are built-in in Spark.